packages/block-library/src/cover/edit.native.js

```diff
@@ -330,27 +348,45 @@ const Cover = ( {
 		<View style={ styles.backgroundContainer }>
 			{ controls }
 
+			{ openMediaOptionsRef?.current && (
+				<View style={ styles.imageEditButton }>
+					<ImageEditingButton
+						isSelected={ isParentSelected }
+						isUploadFailed={ didUploadFail }
+						isUploadInProgress={ isUploadInProgress }
+						onSelectMediaUploadOption={ onSelectMedia }
+						openMediaOptions={ openMediaOptionsRef?.current }
+						url={ url }
+					/>
+				</View>
+			) }
+
 			<View
 				pointerEvents="box-none"
 				style={ [ styles.content, { minHeight: CONTAINER_HEIGHT } ] }
 			>
 				<InnerBlocks template={ INNER_BLOCKS_TEMPLATE } />
 			</View>
 
-			<View pointerEvents="none" style={ overlayStyles }>
-				{ gradientValue && (
-					<Gradient
-						gradientValue={ gradientValue }
-						style={ styles.background }
-					/>
-				) }
+			<View pointerEvents="none" style={ styles.overlayContainer }>
+				<View style={ overlayStyles }>
+					{ gradientValue && (
+						<Gradient
+							gradientValue={ gradientValue }
+							style={ styles.background }
+						/>
+					) }
+				</View>
 			</View>
 
 			<MediaUpload
 				allowedTypes={ ALLOWED_MEDIA_TYPES }
 				isReplacingMedia={ true }
 				onSelect={ onSelectMedia }
-				render={ renderBackground }
+				render={ ( { open, getMediaOptions } ) => {
+					openMediaOptionsRef.current = open;
```

Comment:
If we are keeping `openMediaOptionsRef` in as a ref, do we still need to pass `open` to `renderBackground`?

Reply:
We don't :D I'll update the rest of the code so it uses the ref instead of passing it as a parameter.